\documentclass[12pt]{article}
%%%%%PACKAGES%%%%%
\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}
\usepackage{booktabs}
\usepackage{lmodern}
\usepackage{gensymb}
\usepackage{graphicx}
\usepackage{fancyhdr}
\usepackage{geometry}
\usepackage{listings}
\usepackage{tikz}
\usepackage{color, colortbl}
\usepackage{float}
\usepackage{listings}
\usepackage{hyperref}
\usepackage{subcaption}
\usepackage{geometry}
\usepackage[margin=10pt,font=footnotesize,labelfont=bf,labelsep=space]{caption}

\setlength\unitlength{1mm}
\hypersetup{
	colorlinks,
	linkcolor={blue!50!black},
	citecolor={blue!80!black},
	urlcolor={blue!80!black}
}

\geometry{margin=1.5in}

\definecolor{Gray}{gray}{0.9}
%%%%%%%%%%%%%%%%%%
%%%%%OPTIONS%%%%%%
%%HEADER%%
\pagestyle{fancy}
\lhead{BIO720 Final Project}
\chead{}
\rhead{Yasser Salama}


%%%%%COMMANDS%%%%%
\newcommand{\esal}{\textit{E. salsugenium}}
%%%%%TITLE%%%%%%%%
%Alternate Title%Methodology and Application of Metagenomics for the Characterization of Bacterial Populations in Aquatic Environments
\title{\textsc{Transcriptional Response to Drought Stress in \textit{Eutrema salsugenium}}}
\author{
	{\textsc{\Large Yasser Salama}}\\
	\\
	{McMaster University}\\
	{Hamilton, Ontario, Canada, L8S 4E8}\\
	%{\includegraphics[scale=0.03]{"D:/Dropbox/School/Graduate School/Masters/Thesis/Images/logo_crop"}}
	}
\date{
	{\today}\\
	{\vfill BIO 720 | Bioinformatics}\\
	{Submitted to:}\\
	{Drs. \textsc{Dworkin, Evans, and Golding}}
	}
\begin{document}
	 
	\maketitle
	\thispagestyle{empty}
	\clearpage
	\pagenumbering{roman}
	\tableofcontents
	\listoffigures
	\listoftables
	\clearpage
	\section*{Preface}
	The GitHub repository for this report is located \href{https://github.com/salamayg/BIO_720_final-project}{here.} The repository is set up to contain only the shell, perl, and R scripts used for generating all of the results, as well as the text files containing the count data. The repository does not contain the read files or any subsequent files associated with the reads. The repository is not a clone of the directory on the cluster, however, all the files on the cluster can be accessed (read permission) at /home/yasser/bio720/final\_project. 
	
	When explaining any method or result generated through a script, I will refer to the script in brackets by its names in the scripts folder of the GitHub repository. All the scripts are commented and contain more information about the individual commands than the report will give. 
	\clearpage
	\section{Introduction}
	\pagenumbering{arabic} 
	The prevention of crop loss in the horticulture and agriculture sectors has been a long-standing concern because of the significant economic and societal impacts these losses have. The agriculture industry makes up almost 7\% of Canada's gross domestic product, and with a growing world population, food and water scarcity, and rising global temperatures, these effects will become magnified~\cite{govcan:2015:Online,asseng2015rising,schewe2014multimodel}. 
	
	Abiotic stressors such as drought, salinity, and extreme temperature variations are major drivers of crop loss and can result in 50\% to 80\% loss of crop yield, depending on the crop and growth stage~\cite{boyer1982plant}. Drought is a major facet of abiotic stressors which accounts for approximately 40\% of the indemnities paid for crop loss, larger than any other stressor. As well, in certain crops such as maize and rice, drought alone can account for up to 80\% of the yield loss~\cite{farooq2009plant}. These are extravagant losses with far-reaching impacts. It is therefore imperative to examine the mechanisms by which drought stress can impact plants and their health, and how resistance mechanisms are adapted to such conditions.
	
	The plant response to drought stress is well documented, albeit not fully characterized. In general, the reduction in water availability limits turgidity, inhibiting cell elongation and negatively impacting morphology~\cite{farooq2009plant}. As well, limited nutrient availability, photosynthetic deficits, and consequently diminished mitosis directly impact the health of the plant. 
	The mechanisms by which plants can resist against drought can vary greatly, but typically, there is an attempt to reduce the water requirement through reduced leaf number and surface area, and to maximize water uptake through root proliferation. Physiological responses also result in protection against oxidative stress which arise from the generation of reactive oxygen species. These mechanisms involve elevated expression of enzymatic constituents for oxidative stress protection~\cite{farooq2009plant}. Overall, the stress response and resistance mechanisms are extremely complex, and encompass a variety of biological mechanisms. 
	
	The complexity of the stress response is supplemented by the finding that over-expression of known stress-responsive genes can sometimes have little effect on mitigating the stressors' impact~\cite{bartels2005}. This is likely attributed to the network and co-regulatory nature of the stress response, and requires a more comprehensive understanding of the underlying molecular and physiological changes which occur in response to stress. A useful tool for understanding the stress response and the mechanisms by which it manifests is to examine the complete transcriptional regulation network and the changes that occur once a stress is applied. For the example of drought stress, restriction of water and subsequent examination of the transcriptional changes which occur may provide great insight into the mechanisms for the stress response, and in the case of a drought-tolerant organism, the drought resistance mechanism. 
	
	\textit{Eutrema salsugineum} is a halophytic plant, capable of thriving in various abiotic stress conditions, and is an ideal organism to study the response and resistance mechanism to drought. \esal{} is closely related to the well studied model plant \textit{Arabidopsis thaliana}, with an estimated divergence time of around 43 million years (Figure~\ref{phylo})~\cite{yang2013reference}. An important physiological adaptation in \esal{} is the presence of a low permeability cuticle, which reduces the loss of water through the leaves. This waxy cuticle has been observed to be much more abundant in \esal{} than \textit{A. thaliana}, suggesting that this is an adaptive trait for drought resistance. 
	
		\begin{figure}[H]
			\centering
			\includegraphics[scale=2]{../figures/esal_phylo.jpg}
			\caption[Brassicaceae Phylogeny]{Brassicaceae phylogeny inferred with two markers, \textit{ndhF} and \textit{PHYA}~\cite{yang2013reference}.}
			\label{phylo}
		\end{figure}
		
	Intriguingly, two different ecotypes of \esal{}, Shandong and Yukon, exist, with different standard environmental conditions. Whereas the Yukon ecotype is sustained in sub-arctic temperatures and a semi-arid climate, the Shandong region is more temperate, although both regions feature a high salinity soil. These two ecotypes show distinct responses to drought stress \cite{xu2014leaf}. Previous findings have shown that the Shandong ecotype seems to have a delayed response to drought and differential changes in cuticle permeability traits between the two ecotypes~\cite{macleod2015exposure}. 
	
		\begin{figure}[H]
			\centering
			\includegraphics[scale=0.2]{../figures/esal_morpho.png}
			\caption[Shandong and Yukon ecotype morphology]{Shandong and Yukon accessions of \esal{}~\cite{macleod2015exposure}.}
			\label{morpho}
		\end{figure}
		
	The underlying reasons for the differential stress response and resistance of these two ecotypes is not well characterized, and provides a great opportunity to possibly elucidate the molecular mechanisms through which drought is tolerated in plants. By examining the transcriptional adjustments that \esal{} undergoes in response to stress, we may begin to understand what pathways are important for this process. By examining two different ecotypes, we can also identify common mechanisms and mechanisms unique to each accession, which is quite informative. 
	
	In this work, we examine the transcriptional response of both \esal{} ecotypes to drought through RNA-seq analysis of Shandong and Yukon ecotypes of \esal{} in well-watered and drought treated conditions (experimental design and processing provided by Weretilnyk lab). We also examine how the recovery of these plants manifests through a re-watering phase. 
	
	\section{Methods}
	\label{methods}
	\subsection{Plant Growth and Drought Simulation}
	\esal{} seeds corresponding to Yukon (Y) and Shandong (S) accessions were grown at 4\degree C for 4 and 7 days, respectively, to synchronize germination~\cite{macleod2015exposure}. Subsequently, pots were transferred to growth cabinets simulating standard growth conditions with an optimal watering schedule. The plants were grown under these conditions for 4 weeks after which the plants underwent exposure to drought simulation through complete water-restriction. Effects of the water-restriction were measured gravimetrically using the fraction of transpirable soil water (FTSW) metric. When the fraction reached 0\% (i.e. no remaining water in soil), the plants visibly wilted. Re-watering commenced on the day of wilting to 50\% of the soil capacity, followed by full saturation the next day. Tissue from plants undergoing normal growth conditions (WW, well-watered), drought conditions (D, drought), and re-watering conditions (RW, re-watered) was collected and total RNA was then isolated and processed for sequencing. 
	
	\subsection{Sequencing and RNA-seq Pipeline}
	Two replicates of each treatment and ecotype were used in this analysis, resulting in 12 total samples. Sequencing was completed on a HiSeq 2000 in four batches. A subset of the samples were split onto two lanes, while the rest were run solely on one lane (multiple lane samples are denoted by L1 and L2, see Table \ref{table:samples}). 
	
	FastQC (v0.11.3) was used to validate read metrics prior to and after trimming~\cite{andrews2010fastqc} (script: Step1\_Concat\_trim.sh). A simple loop through all the generated FastQC output files using
	\begin{lstlisting}
	more */summary.txt | grep FAIL
	\end{lstlisting}
	from the fastqc\_reports/pre\_process/ and fastqc\_reports/post\_process/ folders provides a crude way of seeing which metrics fail.
	
	Quality trimming and adapter removal was accomplished with BBDuk from the BBMap suite~\cite{bbmap} (script: Step1\_Concat\_trim.sh). The reference genome and annotation file for \esal{}~\cite{yang2013reference} were downloaded from Phytozyme (v10.3)~\cite{goodstein2012phytozome}. The reference genome is 241 Mb and corresponds to the Shandong ecotype. STAR (v2.3.2) was used to index the reference genome of \esal{} and map the quality filtered reads to the reference~\cite{dobin2013star} (scripts: Step2\_Index\_ref.sh, Step3\_Align.sh) (Figure~\ref{read_stats}). The alignment file was outputted as a bam file already sorted by coordinated through the flag:
	\begin{lstlisting}
	--outSAMtype BAM SortedByCoordinate
	\end{lstlisting}
	
	The bam file generated is in proper format for use with htseq-count and mitigates the need for subsequent sam to bam conversion and sorting. HTSeq-count(v0.6.1) was used to obtain counts for genes in the genome ~\cite{anders2014htseq} using union mode (scripts: Step4\_Count\_union). 
	
	Read statistics for number of reads prior to and after trimming, as well as number of successfully mapped, were obtained (scripts: trim\_stats.sh, mapped\_stats.sh, read\_statistics.R). 

	Counts were imported into R (v3.2.2)~\cite{rsoftware} and differential gene expression analysis was conducted using the DESeq package (v1.10)~\cite{deseq} (script: diff\_gene\_exp.R). All adjusted p-value cutoffs were set to 0.01.
	
	Gene id conversion to gene descriptions was accomplished with the annotation\_info file provided by Phytozyme. Genes of interest from the contrast analysis were put in a file and using grep, were searched for in the annotation file to obtain the gene description. To search for abscisic acid and dehydrin related genes, a simple grep command, 
	\begin{lstlisting}[breaklines=true]
	grep -e abscisic -e Abscisic -e dehydrin -e Dehydrin Esalsugineum_173_v1.0.annotation_info.txt > abscisic_and_dehydrin_genes.txt
	\end{lstlisting}
	 was used. The  gene ids were extracted using a simple cut -f 3 command and then searched for in the DESeq object.
	
	\section{Results}
	\subsection{Sample Information and Read Statistics}
	The samples used in this study are summarized in Table~\ref{table:samples}. Since lane effects may introduce a confounding factor and impact results, all the samples which were run on two lanes were left separated. Another potential confounding factor present in the data is the fact that all the samples were not run together (i.e. run in different batches). Additionally, the only samples run in batch 4 were SRW-1 and SRW-2, which makes it very difficult to rule out whether this batch introduced a systematic variability not due solely to the sample conditions alone. Although running on different lanes may introduce a confounding factor, if all the samples within a batch were evenly divided between lanes, or all samples are run on the same lane, then that effect is essentially negligible. This is the case with these samples. Within every batch that has samples run on two lanes, all the samples in that batch were split between those lanes (e.g. batch 3 and 4). As well, batch 2 samples were all run in one lane. Therefore, this lane effect within each batch was ignored and only the batch effect was dealt with. There is a slight caveat to this; since batch one has two samples run on lane 3 and one sample run on lane 5, there might be a possible lane effect there, but this is ignored.
	
	\begin{table}[H]
		\centering
		\begin{tabular}{lllrrr}
			\toprule
			Sample & Ecotype & Treatment & Replicate & Lane & Batch\\
			\midrule
			\rowcolor{Gray}
			SWW-1 & Shandong & Well-watered & 1 & 2 & 2\\
			SWW-3.L1 & Shandong & Well-watered & 3 & 1 & 3\\
			SWW-3.L2 & Shandong & Well-watered & 3 & 2 & 3\\
			\rowcolor{Gray}
			SD-1 & Shandong & Drought & 1 & 2 & 2\\
			SD-2 & Shandong & Drought & 2 & 2 & 2\\
			\rowcolor{Gray}
			SRW-1.L1 & Shandong & Re-watered & 1 & 1 & 4\\
			\rowcolor{Gray}
			SRW-1.L2 & Shandong & Re-watered & 1 & 2 & 4\\
			SRW-2.L1 & Shandong & Re-watered & 2 & 1 & 4\\
			SRW-2.L2 & Shandong & Re-watered & 2 & 2 & 4\\
			\rowcolor{Gray}
			YWW-1 & Yukon & Well-watered & 1 & 5 & 1\\		
			YWW-2.L1 & Yukon & Well-watered & 2 & 1 & 3\\
			YWW-2.L2 & Yukon & Well-watered & 2 & 2 & 3\\
			\rowcolor{Gray}
			YD-1 & Yukon & Drought & 1 & 3 & 1\\
			YD-2.L1 & Yukon & Drought & 2 & 1 & 3\\
			YD-2.L2 & Yukon & Drought & 2 & 2 & 3\\
			\rowcolor{Gray}
			YRW-1 & Yukon & Re-watered & 1 & 3 & 1\\
			YRW-2.L1 & Yukon & Re-watered & 2 & 1 & 3\\
			YRW-2.L2 & Yukon & Re-watered & 2 & 2 & 3\\

	
			\bottomrule
		\end{tabular}
		\caption[Sample Names and Information]{Sample names and information. Each combination of ecotype and treatment had two replicates, with some samples being run across two lanes. All samples were collectively run in four batches.}
		\label{table:samples}
	\end{table}
	
	The samples averaged 72,653,152 paired reads prior to trimming (Figure~\ref{read_stats}) and FastQC analysis as described in the Methods section suggests that read quality was a ``pass'', even prior to trimming. The only metrics which are considered as ``fail'' by FastQC are ``sequence duplication levels'' and metrics associated with kmer and sequence content. The duplication levels are expected for RNA-seq data as many copies of the same transcript will be present. The kmer and sequence content also seems to be an artifact of the library preparation method, which introduces nucleotide biases at the beginning of the sequence. More importantly, the  measures of sequence quality and adapter contamination pass in all of the samples after trimming.
	
		\begin{figure}[H]
			\centering
			\scalebox{0.7}{\input{trim_stats_nolane.tex}}
			\caption[Read Processing Statistics]{Read processing and mapping. Values are given as pairs of reads. Samples run on multiple lanes are left separated. The mean percentage of trimmed reads which were sucessfully mapped is indicated for each ecotype.}
			\label{read_stats}
		\end{figure}
	
	Quality trimming of the samples confirmed that the data was relatively high quality (only 2.4\% reads lost due to trimming). Further, mapping was relatively successful, with an average of 94\% of trimmed reads successfully mapping. Since the reference genome was generated using a Shandong ecotype plant, I checked whether there was a difference in the percentage of successfully mapped reads between the Shandong and Yukon samples. It might be expected that Shandong samples would show greater percentage of mapped reads, but there was not enough evidence to say that the two ecotypes mapped with different efficiency (p > 0.1). 
	
	
	\subsection{Batch Effects}
	Since the samples for this study were not all run at the same time, there may be a batch effect which introduces bias within the samples. Unfortunately, the batches are not well mixed (i.e. there is not a good representation of different samples within each batch), which makes it hard to delineate the true differences from the batch differences. However, we can still perform a batch effect analysis and use a conservative approach by removing any genes which show differences due to the batch (script: diff\_gene\_exp.R). When comparing all the samples and batches, we see quite a high number of genes showing differential expression due to batch (~23\% with adjusted p-value < 0.1). However, since batch 4 consists of only Shandong re-watered plants, this might be inflating this value because the batch effect is inherently intertwined with the possible effects of the actual variables we are interested in (ecotype and treatment). I generated a subsetted dataset which removes the SRW samples from the analysis, and re-ran the batch effects just to see how much of a difference these samples would make. When the SRW samples (which are the only samples in batch 4) are removed from the analysis, ~1.3\% genes are showing a difference in expression due to batch. This suggests that the effect is likely magnified because batch 4 contains samples of one type. A way to approach this is to remove the ~1.3\% genes showing differential expression from further analysis, and then to be wary of interesting genes which are part of the 23\% of genes found earlier.
	
	Before removal of the trouble genes, we can perform some cluster analysis to get an idea of our samples and crudely visualize whether there is batch effect. Figure~\ref{pca_batch} provides a principal components analysis (PCA) of the samples, labelled by batch. Although the samples are not distinctly clustering due to batch, there is not even mixing between all of the batches either. However, this would be expected due to the lack of mixing between the samples. For instance, if all the samples in one batch correspond to one ecotype, it would not be expected that this batch's samples would cluster with a batch containing only samples of the other ecotype. This is more evident in Figure~\ref{pca_data} where the samples are labelled based on ecotype and treatment. We can see a clear separation of the ecotypes on PC1, and this provides a more reasonable explanation for the clustering pattern seen in Figure~\ref{pca_batch}. PC2 is able to slightly separate the samples based on treatment condition, but the low variance explained suggests that the ecotype is the major factor in the differences between the samples. 


\begin{figure}
	\centering
	\begin{subfigure}[b]{1\textwidth}
		\centering
			\scalebox{0.5}{\input{pca_batch.tex}}
		\caption{}
		\label{pca_batch}
	\end{subfigure}
	~ %add desired spacing between images, e. g. ~, \quad, \qquad, \hfill etc. 
	%(or a blank line to force the subfigure onto a new line)
	\begin{subfigure}[b]{1\textwidth}
		\centering
			\scalebox{0.5}{\input{pca_data.tex}}
		\caption{}
		\label{pca_data}
	\end{subfigure}
	\caption[Cluster analysis with PCA]{Principal components analysis of samples labelled based on \textbf{a)} batch or \textbf{b)} ecotype and treatment}
	\label{pca}
\end{figure}
		
	The samples can also be visualized using a heatmap (Figure~\ref{heatmap}.)	This further presents the separation of the samples based on ecotype, with the Shandong and Yukon ecotype samples clustering together. As well, the SRW samples show a high degree of similarity to each other and form a distinct cluster from the other samples. Again, we cannot be certain of the degree to which batch effects may be playing a role in the separation of the SRW samples, but the pattern of samples clustering together based on the variables of interest and not batch is a good sign. Interestingly, an SWW samples which was run in batch 2 clustered more closely with an SD sample which was also run on batch 2 rather than on the other SWW samples run on batch 3. This hints at the possibility of a batch effect and supports the removal of the ~1.3\% genes discussed earlier.
		
		\begin{figure}[H]
			\centering
			\scalebox{0.7}{\input{heatmap.tex}}
			\caption[Heatmap analysis]{Heatmap analysis. The heatmap shows a measure of simialirity between samples and provides a tree based on the similarity of the samples. The ecotype, treatment, and batch are indicated for each sample.}
			\label{heatmap}
		\end{figure}
	\subsection{Transcriptional Response to Drought}
	Since there are two ecotypes and three treatments, there are a few things to explore in terms of differential gene expression. Firstly, the difference between the Shandong and Yukon ecotype can be examined to see basal differences in gene expression across all treatments (Table~\ref{lfcs} row 1). As well, we can examine the overall effect of drought and re-watering conditions in both ecotypes (rows 2 and 3). More interestingly, however, we can examine the difference in response between the ecotypes to drought or re-watering (Table~\ref{lfcs} rows 5 and 6). These final two comparisons are more informative because a gene may show significant differential expression in response to a treatment in only one ecotype, but not both (see Figure~\ref{count_plot_example}). 
	Although there may be genes which show differential expression in response to drought (row 2), these genes may be called as differentially expressed even when only one ecotype shows a great response while the other does not. 
	\begin{table}[H]
		\centering
		\begin{tabular}{lrr}
			\toprule
			Comparison & LFC > 1 & LFC < 1 \\
			\midrule
			Shandong vs Yukon & 1824 & 1697 \\
			Drought vs Well-watered & 1740  & 2553 \\
			Re-watered vs Well-watered & 364 & 631 \\
			\rowcolor{Gray}
			Shandong vs Yukon response to: && \\
			\rowcolor{Gray}
			Drought & 661 & 269 \\
			\rowcolor{Gray}
			Re-watering & 209 & 313 \\
			%S vs Y & 3262 (13\%) & 3118 (13\%) \\
			%D vs WW & 3458 (14\%) & 3948 (16\%) \\
			%RW vs WW & 1500 (6.2\%) & 1474 (6.1\%) \\
			%D, S vs Y & 1457 (6\%) & 829 (3.4\%) \\
			%RW, S vs Y & 840 (3.5\%) & 896 (3.7\%) \\
			\bottomrule
		\end{tabular}
		\caption[Differentially expressed gene counts]{Differentially expressed gene counts. Number of genes showing log 2 fold changes (LFC) greater than 1 (upregulated) or less than 1 (downregulated) for multiple comparisons at an adjusted p-value cutoff of 0.01. Comparisons are relative to Yukon and well-watered values. Gray coloured rows show the number of genes which showed a different response between Shandong and Yukon in reaction to either treatment.}
		\label{lfcs}
	\end{table}
	
	From Table~\ref{lfcs}, we can see that there are clear differences between the Shandong and Yukon plants in terms of overall gene expression, with Shandong showing about an even number of genes which are upregulated and downregulated compared to Yukon. As well, the response to drought seems to show more downregulation of genes than upregulation, and similarly with the re-watered condition. Despite the clustering of samples seen in Figure~\ref{pca_data}, which seemed to suggest that ecotype had a larger effect on sample similarity than treatment, both ecotype and treatment comparisons show a similar number of differentially expressed genes. This might be explained by the size of the log 2 fold changes within each comparison. Indeed, the absolute mean log 2 fold change in the ecotype comparison was 2.5, in contrast with 1.5	for the drought versus well-watered comparison.
	
%	There is a possibility that genes which are downregualted due to drought persist in that state after re-watering. This could be checked by examining the intersect of genes which are down regulated in both conditions (not done in this analysis). 
	\begin{figure}[H]
		\centering
		\begin{subfigure}[b]{1\textwidth}
			\centering
			\includegraphics[trim=1cm 0 0 0,scale=0.5]{../figures/dww_plot.pdf}
			\caption{Drought vs well-watered}
		\end{subfigure}
		\begin{subfigure}[b]{1\textwidth}
			\centering
			\includegraphics[trim=1cm 0 0 0,scale=0.5]{../figures/sy_d_plot.pdf}
			\caption{Shandong vs Yukon response to drought}
		\end{subfigure}	
		\caption[Expression response plots]{An example of the utility of the interaction term and looking at differential response to a treatment. In this example, the same gene is shown to be differentially expressed in the drought vs well-watered model, but when looking at the interaction term of ecotype and treatment, we can see that the gene is only responsive in the Yukon ecotype and not Shandong.}
		\label{count_plot_example}	
	\end{figure}
		
	To examine how Shandong and Yukon react differently to the drought treatment, I looked at the top 20 genes (based on p-value) which showed differential expression between the two ecotypes to drought (Table~\ref{drought_genes}). Since only 20 genes are looked at here, it is hard to draw general trends about the differences. It would be better to examine the complete list of genes and some form of functional enrichment analysis to categorize what functions are altered. However, we can still see some genes which are known to function in the drought response, such as a glutaredoxin family protein and a LOB domain-containing protein. The glutaredoxin protein fits in line with the idea of oxidative stress being a consequence of drought stress, and has been shown to regulate the plant response to stress~\cite{guo2010tomato}. As well, previous characterization of a LOB-like transcription factor was shown to play a part in salt stress~\cite{ariel2010lob}. Also, the differential expression of the photosystem II reaction center W between the ecotypes is interesting as genes involved in the photosynthetic pathway are known to be regulated in response to drought~\ref{farooq2009plant}. These are potential pathways or mechanisms by which the drought response differs between the two ecotypes, as these are genes which were shown to show greater expression in Shandong than Yukon in response to drought.
	
	\begin{table}[H]
		\centering

		\begin{tabular}{lrl}
			\toprule
			Gene ID & LFC & Gene Description \\
			\midrule
			\rowcolor{Gray}
			Thhalv10028838m & 1.71  & ATP synthase delta-subunit gene                          \\
			\rowcolor{Gray}
			Thhalv10028074m & 3.37   & (bHLH) DNA-binding family protein \\
			Thhalv10026359m & 2.08  & Glutaredoxin family protein                              \\
			\rowcolor{Gray}
			Thhalv10025559m & -3.76 & leucoanthocyanidin dioxygenase                           \\
			Thhalv10019053m & 2.61  & Pathogenesis thaumatin superfamily protein       \\
			\rowcolor{Gray}
			Thhalv10018763m & 3.91  & Galactose oxidase repeat superfamily protein       \\
			Thhalv10018663m & 2.20  & B-box type zinc finger protein with CCT domain           \\
			\rowcolor{Gray}
			Thhalv10017988m & 3.09  & Unknown                                                  \\
			Thhalv10014990m & 2.56  & Unknown                                                  \\
			\rowcolor{Gray}
			Thhalv10014734m & 5.70  & Unknown                                                  \\
			Thhalv10013774m & 7.20    & Eukaryotic aspartyl protease family protein              \\
			\rowcolor{Gray}
			Thhalv10010666m & 4.25  & LOB domain-containing protein 38                         \\
			Thhalv10009301m & -6.47  & myo-inositol oxygenase 1                                 \\
			\rowcolor{Gray}
			Thhalv10009104m & 3.17  & photosystem II reaction center W                         \\
			Thhalv10008980m & 4.67  & Glutaredoxin family protein                              \\
			\rowcolor{Gray}
			Thhalv10008115m & 3.03  & myb-like tf family protein             \\
			Thhalv10007302m & -4.51 & Major facilitator superfamily protein                    \\
			\rowcolor{Gray}
			Thhalv10005532m & 4.27  & basic leucine-zipper 1                                   \\
			Thhalv10004468m & -3.48 & Adenosylmethionine decarbox family protein          \\
			\rowcolor{Gray}
			Thhalv10003191m & -2.87 & dual specificity kinase 1 \\
			\bottomrule                            
		\end{tabular}
				\caption[Differentially expressed genes in response to drought]{Differentially expressed genes between ecotypes in response to drought. log 2 fold change (LFC) values are indicated. Top 20 genes based on p-value are shown.}
				\label{drought_genes}
	\end{table}
	
	Known mechanisms of drought response also include the abscisic acid and dehydrin response~\cite{farrant1996presence}. I wanted to examine whether there was any evidence for these response differing between the two ecotypes. There was only one gene annotated with abscisic or dehydrin response, Thhalv10004906m, a dehydrin family protein, which showed a significant log 2 fold change (p-value < 0.01). This protein had a log 2 fold change of -2.4, indicating that it was expressed greater in the Yukon ecotype than the Shandong ecotype in response to drought. It would be interesting to perform a time series analysis of the various dehydrin proteins between the ecotypes to see whether it is expressed earlier and in greater quantity in the Yukon ecotype, which might explain the delayed response to drought seen in Shandong. Between drought and well-watered samples, there were more abscisic acid and dehydrin related genes that showed significant differential expression, which is expected since these are known regulators of the stress response, although only dehydrin genes showed a positive LFC (increased count in response to drought).
	
	\section{Discussion}
	The ability to detect differentially expressed genes between samples reliably depends heavily on a non-confounding experimental design~\cite{leek2010tackling}. In this dataset, samples were run in different batches and sometimes within separate lanes in the same batch. These factors introduce unwanted variability and this makes it difficult to deduce true differences from differences due to technical variations. In this analysis, I attempted to control for this by removing genes which showed differential gene expression due to batch effect. Another approach to this problem is to put each sample into a unique factor and use that in the formula design of DESeq. In future analysis, the DESeq pipeline should be re-run with this modification. Despite this, the cluster analysis provided in Figures~\ref{pca} and~\ref{heatmap} in combination with the removal of problematic genes provided a foundation with which to identify differentially expressed genes relatively accurately.
	
	The differential expression analysis revealed various genes which showed variability due to ecotype, treatment, and the interaction between these two variables. The importance of using the interaction term is emphasized in Figure~\ref{count_plot_example}, where the same gene show variability due to both ecotype and treatment more so than can be explained by either factor alone. This type of interaction analysis revealed various genes which distinguished the Yukon and Shandong ecotypes when exposed to drought conditions. An important step in providing a more comprehensive analysis of these differences would be an examination of functional enrichment using all the significant differentially expressed genes. This might yield more clues for the discrepancy in drought response between the two ecotypes because looking at a small subset of genes such as in Table~\ref{drought_genes} is very limited. 
	
	The dataset presented in this report is quite extensive. Not all the analysis I wanted to accomplish could be completed for the report, and many things can still be explored. It would be interesting to see the overlap in genes which were similarly regulated in the drought condition and re-watered condition in comparison to the well-watered condition. This would reveal genes which show a more persistent change even when re-introduced to water and may reveal genes which could protect against future droughts. Analyzing the interaction term for ecotype and the re-watering condition would also reveal how the Yukon and Shandong ecotypes might differ in their recovery from drought. This might, for instance, show that one plant can recover to the normal state faster, but the other ecotype might be more resistant for the next drought, at the cost of potentially slower growth or other restrictive mechanism.
	
	This analysis has shown ecotype seems to be a bigger driver of transcriptomic changes than the treatment conditions, although there were many significant differentially expressed genes between treatments and the interaction of ecotype and treatment. As well, genes important in the drought stress response such as dehydrin related genes were differentially expressed in response to drought and also differentially expressed between ecotypes in response to drought. Genes which have a function in mechanisms that are known to be altered in response to drought such as photosynthesis genes and redox genes also showed differential expression between ecotypes. These results highlight the differential response of the two ecotypes to drought and further analysis can aid in understanding the causes for the different response in the ecotypes and understanding the drought resistance mechanism in general.
	
\clearpage
\bibliography{bibliography}
\bibliographystyle{unsrt}
\end{document}


